openhands/core/config/llm_config.py

```diff
@@ -78,6 +79,18 @@ def defaults_to_dict(self) -> dict:
             result[f.name] = get_field_info(f)
         return result
 
+    def __post_init__(self):
+        """
+        Post-initialization hook to assign OpenRouter-related variables to environment variables.
+        This ensures that these values are accessible to litellm at runtime.
+        """
+
+        # Assign OpenRouter-specific variables to environment variables
+        if self.openrouter_site_url:
+            os.environ['OPENROUTER_SITE_URL'] = self.openrouter_site_url
+        if self.openrouter_app_name:
+            os.environ['OPENROUTER_APP_NAME'] = self.openrouter_app_name
```

Comment:
Why use different values now? 🤔 

In litellm's `main.py`:
```py
            openrouter_site_url = get_secret("OR_SITE_URL") or "https://litellm.ai"
            openrouter_app_name = get_secret("OR_APP_NAME") or "liteLLM"

            openrouter_headers = {
                "HTTP-Referer": openrouter_site_url,
                "X-Title": openrouter_app_name,
            }
```
https://github.com/BerriAI/litellm/blob/9a5115d5ea6f7864ec3bd4809f2e0771bb90ebb5/litellm/main.py#L1971-1977

Reply:
Wow that's weird, fixing. Thanks!